AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/SubscriptionSync/ReconcileAndLocalSave/ReconcileAndLocalSaveOperation.swift

```diff
@@ -218,15 +216,18 @@ class ReconcileAndLocalSaveOperation: AsynchronousOperation {
         }
 
         // TODO: Wrap this in a transaction
-        if remoteModel.syncMetadata.deleted {
-            saveDeleteMutation(storageAdapter: storageAdapter, remoteModel: remoteModel)
+        if mutationType == .delete {
+            saveDeleteMutation(storageAdapter: storageAdapter, remoteModel: remoteModel, mutationType: mutationType)
         } else {
-            saveCreateOrUpdateMutation(storageAdapter: storageAdapter, remoteModel: remoteModel)
+            saveCreateOrUpdateMutation(storageAdapter: storageAdapter,
+                                       remoteModel: remoteModel,
+                                       mutationType: mutationType)
         }
-
     }
 
-    private func saveDeleteMutation(storageAdapter: StorageEngineAdapter, remoteModel: RemoteModel) {
+    private func saveDeleteMutation(storageAdapter: StorageEngineAdapter,
+                                    remoteModel: RemoteModel,
+                                    mutationType: MutationEvent.MutationType) {
```

Comment:
This will always be `.delete` right? Any reason we need it passed in?

Reply:
looks like we don't need to, removing